com.unity.ml-agents/Runtime/DemonstrationRecorder.cs

```diff
@@ -11,7 +11,7 @@ namespace MLAgents
     /// </summary>
     [RequireComponent(typeof(Agent))]
     [AddComponentMenu("ML Agents/Demonstration Recorder", (int)MenuGroup.Default)]
-    public class DemonstrationRecorder : MonoBehaviour
+    internal class DemonstrationRecorder : MonoBehaviour
```

Comment:
If we make the class internal, can users still add it to the GameObjects in the editor?

Reply:
Yes (but I don't know why)

Reply:
I think it makes it possible to add the component but not possible to get the component via script (`gameObject.GetComponent<TheComponentIsInternalSoICannotDoThis>()`)

Reply:
Let's talk about it more tomorrow. This feels like the sort of thing that someone might want to access from code; I think we should err on the side of keeping it accessible.

Reply:
Can you hold off on this one for now? I'd like to do a refactor pass on the DemonstrationRecorder/Store separately from this.

Reply:
Reverted